jmxtrans-cluster/src/main/java/com/googlecode/jmxtrans/cluster/exceptions/ClusterException.java

```diff
@@ -0,0 +1,103 @@
+package com.googlecode.jmxtrans.cluster.exceptions;
+
+
+import java.io.PrintStream;
+import java.io.PrintWriter;
+import java.util.Map;
+import java.util.TreeMap;
+
+public class ClusterException extends RuntimeException {
+
+    private static final long serialVersionUID = 1L;
+
+    public static ClusterException wrap(Throwable exception, ErrorCode errorCode) {
+        if (exception instanceof ClusterException) {
+            ClusterException se = (ClusterException)exception;
+            if (errorCode != null && errorCode != se.getErrorCode()) {
+                return new ClusterException(exception.getMessage(), exception, errorCode);
+            }
+            return se;
+        } else {
+            return new ClusterException(exception.getMessage(), exception, errorCode);
+        }
+    }
+
+    public static ClusterException wrap(Throwable exception) {
+        return wrap(exception, null);
+    }
+
+    private ErrorCode errorCode;
+    private final Map<String,Object> properties = new TreeMap<String,Object>();
+
+    public ClusterException(ErrorCode errorCode) {
+        this.errorCode = errorCode;
+    }
+
+    public ClusterException(String message, ErrorCode errorCode) {
+        super(message);
+        this.errorCode = errorCode;
+    }
+
+    public ClusterException(Throwable cause, ErrorCode errorCode) {
+        super(cause);
+        this.errorCode = errorCode;
+    }
+
+    public ClusterException(String message, Throwable cause, ErrorCode errorCode) {
+        super(message, cause);
+        this.errorCode = errorCode;
+    }
+
+    public ErrorCode getErrorCode() {
+        return errorCode;
+    }
+
+    public ClusterException setErrorCode(ErrorCode errorCode) {
+        this.errorCode = errorCode;
+        return this;
+    }
+
+    public Map<String, Object> getProperties() {
+        return properties;
+    }
+
+    @SuppressWarnings("unchecked")
+    public <T> T get(String name) {
+        return (T)properties.get(name);
+    }
+
+    public ClusterException set(String name, Object value) {
+        properties.put(name, value);
+        return this;
+    }
+
+    public void printStackTrace(PrintStream s) {
+        synchronized (s) {
+            printStackTrace(new PrintWriter(s));
+        }
+    }
+
+    public void printStackTrace(PrintWriter s) {
```

Comment:
Overriding `printStackTrace()` to add error code into the stacktrace seems to break the principle of least surprise. You should probably just add that error code to the message.